scripts/preprocess-docs.js

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env node
+
+/**
+ * Pre-processes Mocha's documentation
+ * 1. Update TOC
+ * 2. Dump `--help` output into "Command-Line Usage" section
+ */
+
+/**
+ * The CLI included with markdown-toc doesn't support `bullets`
+ * and `maxdepth` options, so that's why this exists.
+ */
+
+'use strict';
+
+const toc = require('markdown-toc');
+const miss = require('mississippi');
+const {exec} = require('child_process');
+const fs = require('fs');
+const {stripIndents} = require('common-tags');
+const path = require('path');
+
+const docsFilepath = path.join(__dirname, '..', 'docs', 'index.md');
+
+/**
+ * Inserts/updates a table of contents into the documentation
+ * @returns Promise<void>
+ */
+function updateTOC() {
+  return new Promise((resolve, reject) => {
+    console.error(`Updating TOC in ${docsFilepath}...`);
```

Comment:
it doesn't really matter much here, but STDOUT is for program output and STDERR is for logs, errors, etc.

the litmus test is "does it make sense to pipe this to another program?" 

the answer in this case is no, but there's no output otherwise.

Reply:
so `process.stderr.write` is your actual intent...

Reply:
no.. `console.error` which writes to STDERR  is preferable unless you really need direct stream access. among other reasons, if you want to play with control characters (like terminal spinners) or just don’t want the new line console gives you.  but that’s not the case here

Reply:
fwiw  I find `console.error` to be semantically confusing at best. but there’s no alias for it unless we create our own `Console` subclass

Reply:
IMO, just seems poor form to use the semantically named `console.error` for
non-sematic purpose (i.e., to write to `stderr` rather than `stdout` rather than log an error).

Reply:
also this is important to understand:  https://nodejs.org/api/process.html#process_a_note_on_process_i_o

so it's become a habit to use `console.error()` when I want immediate output.

that being said, it doesn't really matter, so I'll just use `console.log`